### PR TITLE
Update doc.go import for staging/src/k8s.io/apimachinery

### PR DESCRIPTION
**Pull request by sanoojm**
What this PR does / why we need it:
update doc.go in:
staging/src/k8s.io/apimachinery

This is necessary so that if someone wants to import only this package, the import path is not referred to as github.com, but as the k8s.io vanity url.
Ref: https://golang.org/doc/go1.4#canonicalimports

Which issue(s) this PR fixes (optional, in fixes #(, fixes #<issue_number>, ...) format, will close the issue(s) when PR gets merged):
Only a partial fix for #68231

Special notes for your reviewer:

Release note:

NONE

### CODE REVIEW

**Review by neolit123 — COMMENTED** [review]
/lgtm
/ok-to-test
/release-note-none
/kind cleanup
/priority backlog

File: staging/src/k8s.io/apimachinery/pkg/util/sets/doc.go [hunk 1]
```diff
@@ -17,4 +17,4 @@ limitations under the License.
 // Code generated by set-gen. DO NOT EDIT.
```
Comment (caesarxuchao) [inline staging/src/k8s.io/apimachinery/pkg/util/sets/doc.go]: You need to change the code generator :)

Comment (sanoojm) [reply staging/src/k8s.io/apimachinery/pkg/util/sets/doc.go]: Done. Thank you!!

**Pull request closed without merging**